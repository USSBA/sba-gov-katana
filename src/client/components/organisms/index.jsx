/* IMPORTS */

// miscellaenous organisms
import BusinessGuideTileCollection from "./business-guide-tile-collection/business-guide-tile-collection.jsx";
import CardCollection from "./card-collection/card-collection.jsx";
import DeveloperTester from "./developer-tester/developer-tester.jsx";
import DocumentCardCollection from "./document-card-collection/document-card-collection.jsx";
import FrontPageHero from "./front-page-hero/front-page-hero.jsx";
import FundingProgramsTileCollection from "./funding-programs-tile-collection/funding-programs-tile-collection.jsx";
import DisasterAlert from "./header-footer/disaster-alert/disaster-alert.jsx";
import NotificationBar from "./header-footer/notification-bar.jsx";
import Hero from "./hero/hero.jsx";
import RelatedDocumentCards from "./related-document-cards/related-document-cards.jsx";
import SearchBox from "./search-box/search-box.jsx";
import SectionNav from "./section-nav/section-nav.jsx";

// header-footer organisms
import Footer from "./header-footer/footer/footer.jsx";
import MobileNav from "./header-footer/mobile-nav/mobile-nav.jsx";
import Header from "./header-footer/header/header.jsx";

// homepage organisms
import Blog from "./homepage/blog/blog.jsx";
import FrontPageLady from "./homepage/front-page-lady/front-page-lady.jsx";
import HappeningNow from "./homepage/happening-now/happening-now.jsx";

// lender match organisms
// -- form
import AdditionalInfoForm from "./lender-match/form/additional-info.jsx";
import BusinessInfoForm from "./lender-match/form/business-info.jsx";
import ContactInfoForm from "./lender-match/form/contact-info.jsx";
import IndustryInfoForm from "./lender-match/form/industry-info.jsx";
import LoanInfo from "./lender-match/form/loan-info.jsx";
import ReviewSection from "./lender-match/form/review-page-helpers.jsx";
import ReviewSubmitInfoForm from "./lender-match/form/review-submit-info.jsx";
// -- landing page
import FindLendersIntro from "./lender-match/landing-page/find-lenders-intro.jsx";
import HelpfulQuestions from "./lender-match/landing-page/helpful-questions.jsx";
import {HowItWorksSection} from "./lender-match/landing-page/how-it-works.jsx";
import PreparationChecklist from "./lender-match/landing-page/preparation-checklist.jsx";
// -- success page
import ConfirmSection from "./lender-match/success-page/confirm-section.jsx";
import DynamicCounselingAndTools from "./lender-match/success-page/counseling-and-tools.jsx";
import CounselorMap from "./lender-match/success-page/counselor-map.jsx";
import EmailConfirmationInvalid from "./lender-match/success-page/email-confirmed-page.jsx";

// modal organisms
import LeaveSbaModal from "./modals/leave-sba-modal/leave-sba-modal.jsx";
import MobileSectionNavModal from "./modals/mobile-section-nav/mobile-section-nav.jsx";

/* EXPORTS */

// miscellaneous organisms
export {
	BusinessGuideTileCollection,
	CardCollection,
	DeveloperTester,
	DocumentCardCollection,
	FrontPageHero,
	FundingProgramsTileCollection,
	DisasterAlert,
	NotificationBar,
	Hero,
	RelatedDocumentCards,
	SearchBox,
	SectionNav
};

// header-footer organisms
export {
	Footer,
	MobileNav,
	Header
};

// homepage organisms
export {
	Blog,
	FrontPageLady,
	HappeningNow
};

// lender match organisms
export {
	AdditionalInfoForm,
	BusinessInfoForm,
	ContactInfoForm,
	IndustryInfoForm,
	LoanInfo,
	ReviewSection,
	ReviewSubmitInfoForm,
	FindLendersIntro,
	HelpfulQuestions,
	HowItWorksSection,
	PreparationChecklist,
	ConfirmSection,
	DynamicCounselingAndTools,
	CounselorMap,
	EmailConfirmationInvalid
};

// modal organisms
export {
	LeaveSbaModal,
	MobileSectionNavModal
};
