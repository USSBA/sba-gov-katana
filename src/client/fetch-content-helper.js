import axios from 'axios'
import path from 'path'
import queryString from 'querystring'

async function fetchRestContent(id, langOverride) {
  let data = null

  try {
    const response = await axios.get(
      '/api/content' + (id ? '/' + id : '') + '.json',
      langOverride && { headers: { 'accept-language': langOverride } }
    )

    if (langOverride === 'es') {
      data = response.data.spanishTranslation
    } else {
      data = response.data
    }
  } catch (error) {
    console.error('fetchRestContent', error)
  }

  return data
}

async function fetchApiDistrictOfficeName(zipcode) {
  let data = null
  try {
    const response = await axios.get(
      `https://h4z2sm4z28.execute-api.us-east-1.amazonaws.com/default/dev-ryan-district-office-zipcode-api?zipcode=${zipcode}`
    )
    data = response.data
  } catch (error) {
    console.error('fetchDistrictOfficeName', error)
  }
  console.log('the data is', data)
  return data
}

async function fetchApiSbaGovContent(fileDirectory) {
  let data = null

  try {
    const response = await axios.get(`https://api.sba.gov/${fileDirectory}`)

    data = JSON.parse(response.data)
  } catch (error) {
    console.error('fetchApiSbaGovContent', error)
  }

  return data
}

async function fetchEventContent(id, langOverride) {
  let data = null

  try {
    const response = await axios.get(
      '/api/content/search/event' + (id ? '/' + id : '') + '.json',
      langOverride && { headers: { 'accept-language': langOverride } }
    )
    data = response.data
  } catch (error) {
    console.error('fetchEventContent', error)
  }

  return data
}

async function fetchSiteContent(type, query, langOverride) {
  let data = null

  try {
    const response = await axios.get(
      '/api/content/search/' + type + '.json' + (query ? '?' + queryString.stringify(query) : ''),
      langOverride && { headers: { 'accept-language': langOverride } }
    )
    data = response.data
  } catch (error) {
    console.error('fetchSiteContent', error)
  }
  return data
}

async function postMiscAction(endpoint, body) {
  const url = path.join('/actions/misc', `${endpoint}`)

  let data = null

  try {
    const response = await axios.post(url, body)
    data = response.data
  } catch (error) {
    console.error('postMiscAction', error)
    throw error
  }

  return data
}

async function runMiscAction(type, query) {
  const url = '/actions/misc/' + type + '.json' + (query ? '?' + queryString.stringify(query) : '')
  let data = null
  try {
    const response = await axios.get(url)
    data = response.data
  } catch (error) {
    console.error('fetchSiteContent', error)
  }
  return data
}

export {
  fetchApiDistrictOfficeName,
  fetchApiSbaGovContent,
  fetchRestContent,
  fetchEventContent,
  fetchSiteContent,
  postMiscAction,
  runMiscAction
}
