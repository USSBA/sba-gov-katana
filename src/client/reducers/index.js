import {combineReducers} from 'redux';
import {routerReducer} from 'react-router-redux';

// our customer reducers
import loanReducer from '../reducers/loan-form.js';
import additionalInfoReducer from '../reducers/additional-info.js';
import contactInfoReducer from '../reducers/contact-info.js';
import businessInfoReducer from '../reducers/business-info.js';
import reviewSubmitInfoReducer from './review-submit-info.js';
import industryInfoReducer from './industry-info.js';

// combine the custom reducers with the routing reducer
const rootReducer = combineReducers({
  loanReducer: loanReducer,
  additionalInfoReducer: additionalInfoReducer,
  contactInfoReducer: contactInfoReducer,
  businessInfoReducer: businessInfoReducer,
  reviewSubmitInfoReducer: reviewSubmitInfoReducer,
  industryInfoReducer: industryInfoReducer,
  routing: routerReducer
});

export default rootReducer;
