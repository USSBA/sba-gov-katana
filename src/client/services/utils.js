import querystring from 'querystring'

function getQueryParams() {
  let queryParams = {}
  const { search } = window.location
  if (search && search !== '') {
    queryParams = querystring.decode(search.replace('?', ''))
  }
  return queryParams
}

export { getQueryParams }
