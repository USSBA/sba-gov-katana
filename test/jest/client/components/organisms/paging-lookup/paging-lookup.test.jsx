/* eslint-disable no-undef */
/*global expect*/

import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import { PagingLookup } from 'organisms'
import * as helper from 'client/fetch-content-helper'
import * as utils from 'client/services/utils.js'

const props = {
  title: 'My Documents',
  type: 'documents',
  taxonomyFilters: ['documentType', 'documentActivity', 'validTaxonomyFilter'],
  fieldsToShowInDetails: ['Type', 'Name', 'Summary'],
  sortByOptions: ['Last Updated', 'Size'],
}

const testTaxonomies = [{
  name: 'documentType',
  terms: [
    'Information notice',
    'Policy Guidance',
    'Policy notice',
    'Procedural notice',
    'Report',
    'SBA form',
    'SOP',
    'Support',
    'TechNote'
  ]}]

describe('PagingLookup', () => {
  var mockGetQueryParams, mockFetchSiteContent
  beforeEach(() => {
    mockGetQueryParams = jest.spyOn(utils, 'getQueryParams')
    mockFetchSiteContent = jest.spyOn(helper, 'fetchSiteContent')
    mockFetchSiteContent.mockReturnValue(testTaxonomies)
  })
  afterEach(() => {
    mockGetQueryParams.mockRestore()
    mockFetchSiteContent.mockRestore()
  })
  test('should render with all the information', () => {
    const component = renderer.create(<PagingLookup {...props} />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('should have an empty default query search', () => {
    mockGetQueryParams.mockReturnValue({})
    const result = shallow(<PagingLookup {...props} />).props().queryState
    expect(result.searchTerm).toBe('')
    expect(result.documentType).toBe('All')
    expect(result.documentActivity).toBe('All')
    expect(result.validTaxonomyFilter).toBe('All')
  })
  test("should accept a 'search' term", () => {
    mockGetQueryParams.mockReturnValue({ search: 'My Search Term' })
    const result = shallow(<PagingLookup {...props} />).first().props().queryState.searchTerm
    expect(result).toBe('My Search Term')
  })
  test("should accept a 'q' term", () => {
    mockGetQueryParams.mockReturnValue({q: 'My Q Search Term' })
    const result = shallow(<PagingLookup {...props} />).first().props().queryState.searchTerm
    expect(result).toBe('My Q Search Term')
  })
  test("should accept 'type' from the query string, converting it to documentType", () => {
    mockGetQueryParams.mockReturnValue({ type: 'SBA Form' })
    const result = shallow(<PagingLookup {...props} />).first().props().queryState.documentType
    expect(result).toBe('SBA Form')
  })
  test("should accept 'activity' from the query string, converting it to documentActivity", () => {
    mockGetQueryParams.mockReturnValue({ activity: 'All' })
    const result = shallow(<PagingLookup {...props} />).first().props().queryState.documentActivity
    expect(result).toBe('All')
  })
  test("should accept 'validTaxonomyFilter' from the query string if in the taxonomyFilter", () => {
    mockGetQueryParams.mockReturnValue({ validTaxonomyFilter: 'My Taxonomy Filter' })
    const result = shallow(<PagingLookup {...props} />).first().props().queryState.validTaxonomyFilter
    expect(result).toBe('My Taxonomy Filter')
  })
  test("should reject 'arbitraryNonsense' from the query string if NOT in the taxonomyFilter", () => {
    mockGetQueryParams.mockReturnValue({ arbitraryNonsense: 'My Arbitrary Nonsense'})
    const result = shallow(<PagingLookup {...props} />).first().props().queryState.arbitraryNonsense
    expect(result).toBeUndefined()
  })
})
