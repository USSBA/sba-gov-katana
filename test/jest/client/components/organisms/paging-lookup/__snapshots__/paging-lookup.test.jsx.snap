// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PagingLookup should render with all the information 1`] = `
<div>
  <div
    className="banner"
  >
    <h2
      className="header"
    >
      My Documents
    </h2>
  </div>
  <div
    className="result"
  >
    <div />
    <div
      className="documentCardCollection"
    >
      <div
        className="emptyDocuments"
      >
        <p
          className="emptyDocumentsMessage"
        >
          Sorry, we couldn't find any 
          documents
           matching that query.
        </p>
        <p>
          <a
            onClick={[Function]}
          >
            Clear all search filters
          </a>
        </p>
      </div>
    </div>
    <div />
  </div>
</div>
`;
